src/code/z_kankyo.c

```diff
@@ -698,12 +1273,399 @@ s32 Environment_IsSceneUpsideDown(PlayState* play) {
+                (s16)(envCtx->lightSettings.light2Color[i] + envCtx->adjLightSettings.light2Color[i]);
+        }
+
+        if ((s16)(envCtx->lightSettings.fogColor[i] + envCtx->adjLightSettings.fogColor[i]) > 255) {
+            lightCtx->fogColor[i] = 255;
+        } else if ((s16)(envCtx->lightSettings.fogColor[i] + envCtx->adjLightSettings.fogColor[i]) < 0) {
+            lightCtx->fogColor[i] = 0;
+        } else {
+            lightCtx->fogColor[i] = (s16)(envCtx->lightSettings.fogColor[i] + envCtx->adjLightSettings.fogColor[i]);
+        }
+    }
+
+    // Set both directional light directions
+    envCtx->dirLight1.params.dir.x = envCtx->lightSettings.light1Dir[0];
+    envCtx->dirLight1.params.dir.y = envCtx->lightSettings.light1Dir[1];
+    envCtx->dirLight1.params.dir.z = envCtx->lightSettings.light1Dir[2];
+
+    envCtx->dirLight2.params.dir.x = envCtx->lightSettings.light2Dir[0];
+    envCtx->dirLight2.params.dir.y = envCtx->lightSettings.light2Dir[1];
+    envCtx->dirLight2.params.dir.z = envCtx->lightSettings.light2Dir[2];
+
+    // Adjust fog near and far if necessary
+    adjustment = envCtx->lightSettings.fogNear + envCtx->adjLightSettings.fogNear;
+
+    if (adjustment <= ENV_FOGNEAR_MAX) {
+        lightCtx->fogNear = adjustment;
+    } else {
+        lightCtx->fogNear = ENV_FOGNEAR_MAX;
+    }
+
+    adjustment = envCtx->lightSettings.zFar + envCtx->adjLightSettings.zFar;
+
+    if (adjustment <= ENV_ZFAR_MAX) {
+        lightCtx->zFar = adjustment;
+    } else {
+        lightCtx->zFar = ENV_ZFAR_MAX;
+    }
+
+    if ((envCtx->dirLight1.params.dir.x == 0) && (envCtx->dirLight1.params.dir.y == 0) &&
+        (envCtx->dirLight1.params.dir.z == 0)) {
+        envCtx->dirLight1.params.dir.x = 1;
+    }
+    if ((envCtx->dirLight2.params.dir.x == 0) && (envCtx->dirLight2.params.dir.y == 0) &&
+        (envCtx->dirLight2.params.dir.z == 0)) {
+        envCtx->dirLight2.params.dir.x = 1;
+    }
+}
+#else
+void Environment_UpdateLights(PlayState* play, EnvironmentContext* envCtx, LightContext* lightCtx);
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_UpdateLights.s")
+#endif
+
+void Environment_UpdateSun(PlayState* play) {
+    f32 temp_f0;
+    s16 temp_a0;
+    u16 phi_v0;
+
+    if (!play->envCtx.sunDisabled) {
+        if (play->envCtx.precipitation[PRECIP_RAIN_CUR] != 0) {
+            Math_SmoothStepToF(&sSunPrimAlpha, 0.0f, 0.5f, 4.0f, 0.01f);
+        } else {
+            Math_SmoothStepToF(&sSunPrimAlpha, 255.0f, 0.5f, 4.0f, 0.01f);
+        }
+
+        if (Environment_IsSceneUpsideDown(play)) {
+            phi_v0 = ((void)0, gSaveContext.save.time) + CLOCK_TIME(12, 0);
+        } else {
+            phi_v0 = ((void)0, gSaveContext.save.time);
+        }
+
+        temp_a0 = phi_v0 - CLOCK_TIME(12, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_UpdateSun.s")
+        if (play->csCtx.state != CS_STATE_IDLE) {
+            // TODO bug?
```

Comment:
Oh we should probably mention at least what the possible bug is

Reply:
I'm not exactly sure what the possible bug is 😂 if I put this there, then I don't remember what it was for

Reply:
Git blame says Roz put it, though I think the bug is the third SmoothStep uses `y` again instead of `z`. (Looks like OoT marks it as a bug)